python/taichi/lang/__init__.py

```diff
@@ -279,15 +280,22 @@ def stat_write(avg):
         f.write(f'time_avg: {avg:.4f}')
 
 
+def is_arch_supported(arch):
```

Comment:
I wonder if in the future, we should move this into C++ side, so that `Program` can be simplified a bit: https://github.com/taichi-dev/taichi/blob/470dc97df22e346ce533f9ece53462dd1c5f597b/taichi/program/program.cpp#L50-L83